src/cmap/connection.ts

```diff
@@ -1232,11 +1228,12 @@ export async function writeCommand(
           agreedCompressor: options.agreedCompressor ?? 'none',
           zlibCompressionLevel: options.zlibCompressionLevel ?? 0
         });
+
   const buffer = Buffer.concat(await finalCommand.toBin());
-  await Promise.race([
-    promisify(connection.socket.write.bind(connection.socket))(buffer),
-    aborted(options.signal)
-  ]);
+
+  const socketWriteFn = promisify(connection.socket.write.bind(connection.socket));
```

Comment:
The callback will be called immediately if there is no buffering or later when the buffer has been written. [Here](https://github.com/nodejs/node/blob/main/lib/internal/streams/writable.js#L696-L702) the callback is invoked a few lines after the drain event

Reply:
Generally, I don't think we can rely on Nodejs source code as a source of truth since we need to support 3+ major node versions and source code is likely to change.  We should rely on documentation and/or confirmation with nodejs contributors.

The docs make me _think_ that you're correct?  Although I'm unclear on what exactly "data is finally written out" means, given that it explicitly uses the terminology in the line above "the entire data was flushed successfully to the kernel buffer".

> Sends data on the socket. The second parameter specifies the encoding in the case of a string. It defaults to UTF8 encoding.
> 
> Returns true if the entire data was flushed successfully to the kernel buffer. Returns false if all or part of the data was queued in user memory. ['drain'](https://nodejs.org/api/net.html#event-drain) will be emitted when the buffer is again free.
> 
> The optional callback parameter will be executed when the data is finally written out, which may not be immediately.